framework/source/class/qx/ui/table/model/Simple.js

```diff
@@ -748,6 +810,362 @@ qx.Class.define("qx.ui.table.model.Simple",
+     */
+    addNumericFilter : function(filter, value1, target)
+    {
+      var temp = null;
+
+      if (qx.lang.Array.contains(this.numericAllowed, filter) && target != null)
+      {
+        if (value1 != null) {
+          temp = [filter, value1, target];
+        }
+      }
+
+      if (temp != null) {
+        this.Filters.push(temp);
+      } else {
+        throw new Error("Filter not recognized: value or target is null!");
+      }
+    },
+
+
+    /**
+     * The addRegex method is used to add a regular expression filter to the
+     * table model.
+     *
+     * @param regex {String}
+     *    The regular expression to match against.
+     *
+     * @param target {String}
+     *    The text value of the column to compare against.
+     *
+     * @param ignorecase {Boolean}
+     *    If true, the regular expression will ignore case.
+     *
+     *
+     * @throws {Error} If the regex is not valid.
+     */
+    addRegex : function(regex, target, ignorecase)
+    {
+      var regexarg;
+      if (ignorecase) { regexarg ='gi'; } else { regexarg ='g'; }
+      if (regex != null && target != null) {
+        var temp = new Array("regex", regex, target, regexarg);
+      }
+
+      if (temp != null) {
+        this.Filters.push(temp);
+      } else {
+        throw new Error("regex cannot be null!");
+      }
+    },
+
+
+    /**
+     * The addNotRegex method is used to add a regular expression filter to the
+     * table model and filter cells that do not match.
+     *
+     * @param regex {String}
+     *    The regular expression to match against.
+     *
+     * @param target {String}
+     *    The text value of the column to compare against.
+     *
+     * @param ignorecase {Boolean}
+     *    If true, the regular expression will ignore case.
+     *
+     *
+     * @throws {Error} If the regex is null.
+     */
+    addNotRegex : function(regex, target, ignorecase)
+    {
+      var regexarg;
+      if (ignorecase) { regexarg ='gi'; } else { regexarg ='g'; }
+      if (regex != null && target != null) {
+        var temp = new Array("notregex", regex, target, regexarg);
+      }
+
+      if (temp != null) {
+        this.Filters.push(temp);
+      } else {
+        throw new Error("notregex cannot be null!");
+      }
+    },
+
+
+    /**
+     * The applyFilters method is called to apply filters to the table model.
+     */
+    applyFilters : function()
+    {
+      var i;
+      var filter_test;
+      var compareValue;
+      var rowArr = this.getData();
+      var rowLength = rowArr.length;
+      var count;
+      var hidden = 0;
+      var rowsToHide = [];
+
+      for (var row = 0; row < rowLength; row++) {
+        filter_test = false;
+
+        for (i in this.Filters) {
+
+          if (qx.lang.Array.contains(this.numericAllowed, this.Filters[i][0]) && filter_test === false) {
+            compareValue = this.getValueById(this.Filters[i][2], row);
+
+            switch(this.Filters[i][0]) {
```

Comment:
since `this.Filters[i][0]` does not depend on the `row` the decision should be outside of the potentially long loop over the rows of the table